### PR TITLE
B-19780 INT CSV Export for Queue Tables

### PR DESCRIPTION
**Pull request by ajlusk**
## Summary

Is there anything you would like reviewers to give additional scrutiny?
- Between the duplication for the HQ role and the addition of the exportValue function to the columns arrays in each queue component I've been thinking it might make sense to move those to their own utility/shared file. Thoughts?

[This github post](https://github.com/react-csv/react-csv/issues/72#issuecomment-421160842) explains why the `<CSVLink/>` component is programmatically clicked. That was the best solution I could find with this library to fetch the data asynchronously and still set the file name.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the office app
2. Login as a Counselor (ideally in KKFA due to available test data)
3. Locate the "Export to CSV" button to the top right of your queue table, press it and make sure it exports a CSV that matches the data in the table. The file should contain every move in your queue, not just those on the current page.
4. Apply your choice of filters or sorting, export to CSV again and ensure the data exported matches with applied filter and/or sorting.
5. Repeat for the SC's PPM Closeout Queue.
6. Repeat for the TOO's Queue.
7. Repeat for the TIO's Queue.
8. Repeat for all queues visible to the HQ user.

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

**Comment by ajlusk** [comment]
> Warnings
> ⚠️	This PR does not include changes to storybook, even though it affects component code.
> Generated by 🚫 [dangerJS](https://danger.systems/js) against [cce0487](https://github.com/transcom/mymove/commit/cce0487201798d1e15d200a87a7d72cdf5fead9d)

Resolved

**Review by CoryKleinjanCACI — APPROVED** [review]
Changes look good and tests pass.

File: src/pages/Office/HeadquartersQueues/HeadquartersQueues.jsx [hunk 1]
```diff
@@ -53,19 +59,25 @@ import ConnectedFlashMessage from 'containers/FlashMessage/FlashMessage';
...
-    return null;
-  }),
+  createHeader('ID', 'id', { id: 'id' }),
+  createHeader(
+    ' ',
+    (row) => {
+      const now = new Date();
+      // this will render a lock icon if the move is locked & if the lockExpiresAt value is after right now
+      if (row.lockedByOfficeUserID && row.lockExpiresAt && now < new Date(row.lockExpiresAt) && moveLockFlag) {
+        return (
```
Comment (r-mettler) [inline src/pages/Office/HeadquartersQueues/HeadquartersQueues.jsx:69]: I could see the lock icon being in a utility library, there is a lot of duplication in the files, especially the HQ queue that we could benefit from just adding this in using a utility call.

File: src/pages/Office/HeadquartersQueues/HeadquartersQueues.jsx [hunk 2]
```diff
@@ -81,11 +93,17 @@ const tooQueueColumns = (moveLockFlag, showBranchFilter = true) => [
     {
       id: 'lastName',
       isFilterable: true,
+      exportValue: (row) => {
```
Comment (r-mettler) [inline src/pages/Office/HeadquartersQueues/HeadquartersQueues.jsx:96]: I am fine with just leaving the exportValue as they are.

**Review by r-mettler — APPROVED** [review]
Code looks good. All queue exports worked as expected. Verified Storybook component is working as well. Optional if you want to move the lock icon stuff to a utility library since it is used in all the queues.

**Pull request merged**